mh/pat_item.py

```diff
@@ -585,6 +585,51 @@ class UserSearchInfo(PatData):
         (0x10, "info_mine_0x10"),
     )
 
+class LayerPath(object):
+    FMT_WRITE = struct.Struct(">IIHHH")
+
+    def __init__(self, server_id=None, gate_id=None, city_id=None):
+        # type: (int|None, int|None, int|None) -> None
+        self.server_id = server_id if server_id else 0
+        self.gate_id = gate_id if gate_id else 0
+        self.city_id = city_id if city_id else 0
```

Comment:
You should enforce the `if` check using `is not None`. If these id can't be zero _(I don't think they can)_ then you can get rid of the ternary and use the `or` operator instead (ex: `server_id = server_id or 0`).